GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -539,6 +536,7 @@ void onDrawEyeView(int eye, float fovDegrees) {
 
     /** Called once per frame, before {@link #onDrawEyeView(int, float)}. */
     void onDrawFrame() {
+        mActivity.setCameraRig(mMainScene.getMainCameraRig());
         if (mCurrentEye == 1) {
```

Comment:
Imho, I don't think you will have to setCameraRig for every new frame. I think you'd just need to set the camerarig in GVRViewManager.onSurfaceCreated() after new GVRScene(this) is called(for splashscreen). One more place you need to set is in the GVRViewManager.setMainScene() method(for every new main scene except splashscreen).

I tried above and it works for me.